stdlib/LinearAlgebra/src/LinearAlgebra.jl

```diff
@@ -433,21 +433,56 @@ function versioninfo(io::IO=stdout)
     println(io, "LAPACK: ",Base.liblapack_name)
 end
 
-function __init__()
-    try
-        BLAS.check()
-        if BLAS.vendor() === :mkl
-            ccall((:MKL_Set_Interface_Layer, Base.libblas_name), Cvoid, (Cint,), USE_BLAS64 ? 1 : 0)
+function get_blas_lapack_path()
+    shlib_ext = if Sys.iswindows()
+         ".dll"
+    elseif Sys.isapple()
+         ".dylib"
+    else
+         ".so"
+    end
+    libblas_path = joinpath(Sys.BINDIR, Base.LIBDIR, "julia", string(Base.libblas_name, shlib_ext))
```

Comment:
I think we want this whole method to just return `Base.libblas_name`.  If we want an absolute path in here, then `Base.libblas_name` should already be an absolute path.  Of course, usually we don't want an actual absolute path, we want a fully-specified relative path, which we can do by using [`@executable_path`](https://github.com/JuliaLang/julia/blob/0104471f5d485d8d679d7f62112d90209290b4dc/base/libdl.jl#L106-L109) which is a new feature that I added to `dlopen()` recently.  So we can have `Make.inc` default to setting `LIBBLASNAME` not to just `libopenblas`, but instead to `@executable_path/$(libdir_rel)/libopenblas.so`.  That way it's an unambiguous path passed in to `dlopen()` (so we won't accidentally open some other `libopenblas.so` that happens to be on our `LD_LIBRARY_PATH`), but still relative to the Julia install (so we're still relocatable) _and_ if the user actually sets `LIBBLASNAME`, they can set it to something like `libblas` and `dlopen()` will "just work".

I don't think we need any of the `isfile()` checks in here. We will trust the path, and fail if it doesn't work.  I see here you're trying to cover over the fact that `libblastrampoline` lives in `lib/` during the in-tree build, and then gets moved to `lib/julia` after install.  This is an awkwardness that will go away by the time v1.7 actually launches, since everything will live in stable `artifacts/<hash>` directories.  We can't just modify the `build_h.jl` file after installing it, since we'd have to then rebuild the entire system image.

We have a few options:

1) Wait until stdlibs-as-artifacts lands, then the path will be stable and no rebuilds will be necessary.

2) Wait until `Preferences.jl` lands as an stdlib, use that to provide this information instead of `build_h.jl`, so that we can change it when installing Julia and we don't have to rebuild the system image.

3) Continue to do this `isfile(path)` check and if it fails we insert `julia` into the path.

Reply:
Example of (3):

```
# In Make.inc
```

```
# If this is trying to do a relative lookup in our Julia lib directory
if startswith(libblas_name, joinpath("@executable_path", Base.LIBDIR))
    # If that file doesn't actually exist, insert "julia"
    if !isfile(replace(libblas_name, "@executable_path" => Sys.BINDIR))
        libblas_name = replace(libblas_name, joinpath("@executable_path", Base.LIBDIR) => joinpath(Sys.BINDIR, Base.LIBDIR, "julia"))
    end
end
```

Reply:
I propose that we keep the solution we have for now (which should work for packagers and people building from source), and migrate to Preferences.jl in the 1.7 timeframe.